jfr-daemon/src/main/java/com/newrelic/jfr/daemon/JFRUploader.java

```diff
@@ -82,6 +68,10 @@ private void bufferFileData(Path dumpFile) {
   }
 
   private void maybeDrainAndSend() {
+    if (!readinessCheck.get()) {
+      logger.warn("Drain attempt skipped -- readiness check not yet ready.");
```

Comment:
What do you think about changing this message to say specifically that the entity guid is not available yet?

Reply:
I dislike it because it introduces a fragile text-based dependency that could be easily invalidated later.  

The `JFRUploader` is completely decoupled from the implementation of what it means to be ready.  This class does not concern itself what the details of what it means to be ready....and so it doesn't know anything about an entity guid.